canal/sync.go

```diff
@@ -235,6 +235,14 @@ func parseStmt(stmt ast.StmtNode) (ns []*node) {
 			table: t.Table.Name.String(),
 		}
 		ns = []*node{n}
+	case *ast.AnalyzeTableStmt:
+		ns = make([]*node, len(t.TableNames))
```

Comment:
I see that with this change, `OnTableChanged`, `OnDDL` of handler will be called. My concern is

1. ANALYZE will not change the table. I'm not a user of handler interface but I just read the comment of `OnTableChanged`, it said "You need to clear the associated data like cache with the table." I think with this PR ANALYZE will trigger unnecessary cache dropping.
2. ANALYZE is not a DDL, MySQL put it under "Database Administration Statements - Table Maintenance Statements" https://dev.mysql.com/doc/refman/8.4/en/analyze-table.html. 
 
So I recommend we extend the `EventHandler` interface to add a new callback for it?

Reply:
What about just setting:
```
savePos = true 
force = true 
```

.. regardless of the len(nodes)?

This would solve my problem since I just need the `OnPosSynced` event to fire.

Reply:
> since I just need the OnPosSynced event to fire.

I think we can let `parseStmt` returns a `savePos`, and if any stmt has a `true` `savePos` we can enter https://github.com/go-mysql-org/go-mysql/blob/3b1dd0e7703815062f7999fd7d503df0061807a9/canal/sync.go#L173-L177